lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -392,9 +450,13 @@ def do_3d_projection(self):
         """
         Project the points according to renderer matrix.
         """
+        segments = self._segments3d
+        if self._axlim_clip:
+            segments = [np.ma.column_stack([*_viewlim_mask(*zip(*points), self.axes)])
```

Comment:
Updated to a vectorized version. It's 4 lines now instead of 1, but makes things mostly vectorized.

Reply:
To be clear, I did mean a vectorized version of `_viewlim_mask`, but I don't remember now if it would be useful in all cases.

Reply:
Ah, I see what you mean. I do think any potential overhead is minimal in vectorizing `_viewlim_mask` per-axis as it is right now, rather than casting everything to a single numpy array.